src/main/java/net/opentsdb/tsd/KafkaRpcPluginThread.java

```diff
@@ -273,52 +277,69 @@ public void run() {
           case RAW:
           case ROLLUP:
             // Deserialize the event from the received (opaque) message.
+            List<TypedIncomingData> eventList = new ArrayList<TypedIncomingData>();
             TypedIncomingData event;
             try {
-              event = JSON.parseToObject(message.message(), 
-                  TypedIncomingData.class);
+              event = JSON.parseToObject(message.message(), TypedIncomingData.class);
+              eventList.add(event);
             } catch (Throwable ex) {
```

Comment:
Do we need to catch Throwable here? I think JSON throws an IllegalArgumentException when you try to parse with a wrong class.

Reply:
If like you'd like me to update this to IllegalArgumentException that's not a problem, but just let me know. I was keeping the exception being caught the same so as to not change more than was necessary.